File: app/src/main/java/ru/melod1n/vk/api/UserConfig.java
package ru.melod1n.vk.api;

import android.text.TextUtils;

import ru.melod1n.vk.common.AppGlobal;

public class UserConfig {

    private static final String TOKEN = "token";
    private static final String USER_ID = "user_id";

    public static final String API_ID = "6964679";

    private static String token;
    private static int userId;

    public UserConfig() {
    }

    public UserConfig(String token, int userId) {
        UserConfig.token = token;
        UserConfig.userId = userId;
    }

    public static String getToken() {
        return token;
    }

    public static void setToken(String token) {
        UserConfig.token = token;
    }

    public static int getUserId() {
        return userId;
    }

    public static void setUserId(int userId) {
        UserConfig.userId = userId;
    }

    public static void save() {
        AppGlobal.preferences.edit()
                .putString(TOKEN, token)
                .putInt(USER_ID, userId)
                .apply();
    }

    public static void restore() {
        token = AppGlobal.preferences.getString(TOKEN, "");
        userId = AppGlobal.preferences.getInt(USER_ID, -1);
    }

    public static boolean isLoggedIn() {
        return userId > 0 && !TextUtils.isEmpty(token);
    }
}

File: app/src/main/java/ru/melod1n/vk/api/UserConfig.kt
package ru.melod1n.vk.api

import android.text.TextUtils
import ru.melod1n.vk.common.AppGlobal

object UserConfig {
    private const val TOKEN = "token"
    private const val USER_ID = "user_id"

    const val API_ID = "6964679"

    var token: String? = null
    var userId = 0

    fun save() {
        AppGlobal.preferences.edit().putString(TOKEN, token).putInt(USER_ID, userId).apply()
    }

    fun restore() {
        token = AppGlobal.preferences.getString(TOKEN, "")
        userId = AppGlobal.preferences.getInt(USER_ID, -1)
    }

    fun clear() {
        token = ""
        userId = -1
        AppGlobal.preferences.edit().remove(TOKEN).remove(USER_ID).apply()
    }

    val isLoggedIn: Boolean
        get() = userId > 0 && !TextUtils.isEmpty(token)
}